src/main/java/com/sforce/cd/apexUnit/client/utils/ApexClassFetcherUtils.java

```diff
@@ -321,8 +322,8 @@ public static String fetchAndAddToMapApexClassIdBasedOnName(PartnerConnection co
 		}
 		if (queryResult != null && queryResult.getDone()) {
 			String[] classIds = fetchApexClassesAsArray(queryResult);
-			for (String classId : classIds) {
-				apexClassId = classId;
+			if(classIds.length>0){
```

Comment:
@adarsh-ramakrishna-sfdc  Yes, thought about this case but saw that classids cannot be null because [this](https://github.com/forcedotcom/ApexUnit/blob/namespace-to-query/src/main/java/com/sforce/cd/apexUnit/client/utils/ApexClassFetcherUtils.java#L283) line is always true as we call the method when the same condition is true. Having a unit test case for this helps.

Also found out that, QueryResult.getRecords() does not return null but returns empty array when QueryResult object is created. Correct me if I am wrong. 

[Here](https://gist.github.com/pmedapuram/80298aaadebde36185a8) is the code I tried. A empty object accessing getRecords, and I received true.